File: mobile/assets/create-icon.py
#!/usr/bin/env python3
import os
from PIL import Image, ImageDraw

# Create a 1024x1024 black background
size = 1024
background = Image.new('RGB', (size, size), 'black')

# Load the logo
logo_path = 'logo-square.png'
if os.path.exists(logo_path):
    logo = Image.open(logo_path)
    
    # Resize logo to be smaller (about 60% of the icon size)
    logo_size = int(size * 0.6)
    logo = logo.resize((logo_size, logo_size), Image.Resampling.LANCZOS)
    
    # Center the logo on the black background
    x = (size - logo_size) // 2
    y = (size - logo_size) // 2
    
    # Paste the logo onto the black background
    background.paste(logo, (x, y), logo if logo.mode == 'RGBA' else None)
    
    # Save the result
    background.save('logo.png', 'PNG')
    print("Created new logo with black background and smaller size")
else:
    print("Logo file not found")

File: mobile/assets/create-icon.js
const fs = require('fs');
const { createCanvas, loadImage } = require('canvas');

async function createIcon() {
    try {
        // Create a 1024x1024 canvas with black background
        const canvas = createCanvas(1024, 1024);
        const ctx = canvas.getContext('2d');
        
        // Fill with black background
        ctx.fillStyle = 'black';
        ctx.fillRect(0, 0, 1024, 1024);
        
        // Load and resize the logo
        const logo = await loadImage('logo-square.png');
        const logoSize = 1024 * 0.6; // 60% of icon size
        const x = (1024 - logoSize) / 2;
        const y = (1024 - logoSize) / 2;
        
        // Draw the logo centered
        ctx.drawImage(logo, x, y, logoSize, logoSize);
        
        // Save the result
        const buffer = canvas.toBuffer('image/png');
        fs.writeFileSync('logo.png', buffer);
        
        console.log('Created new logo with black background and smaller size');
    } catch (error) {
        console.error('Error:', error.message);
    }
}

createIcon();
